app/src/forms/auth/middleware/apiAccess.js

```diff
@@ -6,40 +6,99 @@ const formService = require('../../form/service');
+
+  // check to see that an associated submissionId exists
+  if (file && !file.formSubmissionId) {
+    throw new Problem(500, { detail: `Submission ID missing in file storage "${fileId}".` });
+  }
+
+  return _getFormIdFromSubmissionId(file.formSubmissionId);
+};
+
+/**
+ * Gets the Form ID that corresponds to a given Submission ID.
+ *
+ * @param uuid formSubmissionId that has a corresponding Form ID.
+ * @returns a UUID for form that the Submission ID belongs to.
+ * @throws Problem if there is a UUID validation error.
+ */
+const _getFormIdFromSubmissionId = async (formSubmissionId) => {
+  if (!uuidValidate(formSubmissionId)) {
+    throw new Problem(400, { detail: `Bad formSubmissionId "${formSubmissionId}".` });
+  }
+
+  const result = await submissionService.read(formSubmissionId);
+
+  return result?.form?.id;
+};
+
 module.exports = async (req, res, next) => {
   try {
     // Check if authorization header is basic auth
     if (req.headers && req.headers.authorization && req.headers.authorization.startsWith('Basic ')) {
       // URL params should override query string params of the same attribute
       const params = { ...req.query, ...req.params };
 
-      // Basic auth is currently only used for form and submission endpoints. Use
-      // the formId if it exists, otherwise fetch the formId from the submission's
-      // form.
-      let formId;
-      if (params.formId) {
-        formId = params.formId;
-      } else if (params.formSubmissionId && uuidValidate(params.formSubmissionId)) {
-        const result = await submissionService.read(params.formSubmissionId);
-        formId = result?.form?.id;
-      } else if (params.id && uuidValidate(params.id)) {
-        // check for file id (saved as id), get submissionID from request body
-        const sid = await fileService.read(params.id);
-        //check to see that an associated submissionId exists
-        if (!sid || !sid.formSubmissionId) {
-          throw new Error('Submission ID not found in file storage.');
-        }
-        const result = await submissionService.read(sid.formSubmissionId);
-        formId = result?.form?.id;
+      // Get the form ID using whatever parameter we have in the request (Form
+      // ID, Form Submission ID, or File Upload ID).
+      const formId = await _getFormId(params);
+      if (!formId) {
+        throw new Problem(401, { detail: HTTP_401_DETAIL });
       }
 
-      let secret = ''; // Must be initialized as a string
-
-      if (formId && uuidValidate(formId)) {
-        const result = await formService.readApiKey(formId);
-        secret = result && result.secret ? result.secret : '';
+      const apiKey = await formService.readApiKey(formId);
+      if (!apiKey || !apiKey.secret) {
+        throw new Problem(401, { detail: HTTP_401_DETAIL });
       }
 
+      // if (params.id && apiKey.filesApiAccess === false) {
+      //   throw new Problem(401, { detail: HTTP_401_DETAIL });
+      // }
```

Comment:
Code for Jai to duplicate what he has in PR1225 - although this should be a 403 Problem like in PR1225 (oops)!